pkg/apiserver/apiserver_test.go

```diff
@@ -159,159 +147,182 @@ func generateTestKey() (*rsa.PrivateKey, error) {
+	app := &cdiAPIApp{}
+	app.composeUploadTokenAPI()
 
-	signingKeySecret, err := keystest.NewPrivateKeySecret("cdi", apiSigningKeySecretName, signingKey)
-	if err != nil {
-		t.Errorf("error creating secret: %v", err)
-	}
+	req, _ := http.NewRequest("GET", url, nil)
+	rr := httptest.NewRecorder()
 
-	kubeobjects := []runtime.Object{}
-	kubeobjects = append(kubeobjects, signingKeySecret)
+	app.container.ServeHTTP(rr, req)
 
-	actions := []core.Action{}
-	actions = append(actions, signingKeySecretGetAction())
+	status := rr.Code
+	Expect(status).To(Equal(http.StatusOK))
 
-	client := k8sfake.NewSimpleClientset(kubeobjects...)
+	return rr
+}
 
-	app := &cdiAPIApp{
-		client: client,
+func getExpectedAPIGroup() metav1.APIGroup {
+	return metav1.APIGroup{
+		Name: "upload.cdi.kubevirt.io",
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "APIGroup",
+			APIVersion: "v1",
+		},
+		PreferredVersion: metav1.GroupVersionForDiscovery{
+			GroupVersion: "upload.cdi.kubevirt.io/v1alpha1",
+			Version:      "v1alpha1",
+		},
+		Versions: []metav1.GroupVersionForDiscovery{
+			{
+				GroupVersion: "upload.cdi.kubevirt.io/v1alpha1",
+				Version:      "v1alpha1",
+			},
+		},
+		ServerAddressByClientCIDRs: []metav1.ServerAddressByClientCIDR{
+			{
+				ClientCIDR:    "0.0.0.0/0",
+				ServerAddress: "",
+			},
+		},
 	}
+}
 
-	err = app.getKeysAndCerts()
-	if err != nil {
-		t.Errorf("error creating upload api app: %v", err)
-	}
+var _ = Describe("API server tests", func() {
+	It("Get self-signed cert", func() {
+		signingKey, err := generateTestKey()
+		Expect(err).ToNot(HaveOccurred())
 
-	checkActions(actions, client.Actions(), t)
-}
+		signingKeySecret, err := keystest.NewPrivateKeySecret("cdi", apiSigningKeySecretName, signingKey)
+		Expect(err).ToNot(HaveOccurred())
 
-func TestShouldGenerateCertsAndKeyFirstRun(t *testing.T) {
-	kubeobjects := []runtime.Object{}
+		kubeobjects := []runtime.Object{}
+		kubeobjects = append(kubeobjects, signingKeySecret)
 
-	client := k8sfake.NewSimpleClientset(kubeobjects...)
+		actions := []core.Action{}
+		actions = append(actions, signingKeySecretGetAction())
 
-	app := &cdiAPIApp{
-		client: client,
-	}
+		client := k8sfake.NewSimpleClientset(kubeobjects...)
 
-	err := app.getKeysAndCerts()
-	if err != nil {
-		t.Errorf("error creating upload api app: %v", err)
-	}
+		app := &cdiAPIApp{
+			client: client,
+		}
 
-	actions := []core.Action{}
-	actions = append(actions, signingKeySecretGetAction())
-	actions = append(actions, cdiConfigGetAction())
-	actions = append(actions, signingKeySecretCreateAction(app.privateSigningKey))
+		err = app.getKeysAndCerts()
+		Expect(err).ToNot(HaveOccurred())
 
-	checkActions(actions, client.Actions(), t)
-}
+		checkActions(actions, client.Actions())
+	})
 
-func TestGetAPIResouceList(t *testing.T) {
-	rr := doGetRequest(t, "/apis/upload.cdi.kubevirt.io/v1alpha1")
+	It("Should generate certs and key on first run", func() {
+		kubeobjects := []runtime.Object{}
```

Comment:
Don't think we need the kubeobjects here as we are not adding anything, you can just call: client := k8sfake.NewSimpleClientset() instead